src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -0,0 +1,314 @@
+		uintptr(unsafe.Pointer(&bufferLength)),
+		uintptr(unsafe.Pointer(nil)),
+		uintptr(unsafe.Pointer(nil)),
+	)
+	switch rv {
+	case uintptr(windows.ERROR_SUCCESS):
+		break
+	case uintptr(windows.ERROR_INSUFFICIENT_BUFFER):
+		// This is expected: we didn't provide any buffer
+		break
+	default:
+		return nil, errorFromWin32("error getting buffer size", rv, err)
+	}
+
+	buf := make([]byte, bufferLength)
+	rv, _, err = getPackageInfo.Call(
+		packageInfoReference,
+		uintptr(PACKAGE_INFORMATION_BASIC),
+		uintptr(unsafe.Pointer(&bufferLength)),
+		uintptr(unsafe.Pointer(unsafe.SliceData(buf))),
+		uintptr(unsafe.Pointer(&count)),
+	)
+	if rv != uintptr(windows.ERROR_SUCCESS) {
+		return nil, errorFromWin32("error getting package info", rv, err)
+	}
+	infos := unsafe.Slice((*packageInfo)(unsafe.Pointer(unsafe.SliceData(buf))), count)
+	for _, info := range infos {
+		// `info` is a pointer to an unsafe slice; make a copy of the version
+		// on the stack and then return that so the GC knows about it.
+		versionCopy := info.packageId.version
+		return &versionCopy, nil
+	}
+
+	return nil, fmt.Errorf("no info found for %s", fullName)
+}
+
+// runWSLExe runs WSL.exe and returns the standard output.
+// This can be replaced for testing.
+func runWSLExe(ctx context.Context, args ...string) (string, error) {
+	systemDir, err := windows.GetSystemDirectory()
+	if err != nil {
+		return "", fmt.Errorf("failed to get system directory: %w", err)
+	}
+	wslPath := filepath.Join(systemDir, "wsl.exe")
+	cmd := exec.CommandContext(ctx, wslPath, "--status")
+	cmd.Stderr = os.Stderr
+	stdout, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to get WSL status: %w", err)
+	}
+	output := windows.UTF16PtrToString(
+		(*uint16)(unsafe.Pointer(unsafe.SliceData(append(stdout, 0, 0)))),
+	)
+	return output, nil
+}
+
+// isInboxWSLInstalled checks if the "in-box" version of WSL is installed,
+// returning whether it's installed, and whether the kernel is installed
+func isInboxWSLInstalled(ctx context.Context) (bool, bool, error) {
+	runWSLExeFunc := runWSLExe
+	if f := ctx.Value(&kWSLExeOverride); f != nil {
+		runWSLExeFunc = f.(func(context.Context, ...string) (string, error))
+	}
+
+	output, err := runWSLExeFunc(ctx, "--status")
+	var exitErr *exec.ExitError
+	if errors.As(err, &exitErr) && exitErr.ExitCode() == wslExitNotInstalled {
+		return false, false, nil
+	} else if err != nil {
+		logrus.WithError(err).Trace("wsl.exe --status exited")
+		return false, false, err
+	}
+	lines := strings.Split(strings.TrimSpace(output), "\n")
+	if len(lines) < 1 {
+		return false, false, fmt.Errorf("no output from wsl --status")
+	}
+
+	// Check if the kernel is installed.
+	upgradeCodeString := kMsiUpgradeCode
+	if v := ctx.Value(&kUpgradeCodeOverride); v != nil {
+		upgradeCodeString = v.(string)
+	}
+	upgradeCode, err := windows.UTF16PtrFromString(upgradeCodeString)
+	if err != nil {
+		return false, false, err
+	}
+	productCode := make([]uint16, 39)
```

Comment:
nit; optionally, it would be nice to have a constant for the 39. I'm surprised Linter didn't complaint about the magic number 🤔